static/js/components/AssignmentForm.jsx

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { useForm, Controller } from 'react-hook-form';
+import Select from "@material-ui/core/Select";
+import InputLabel from "@material-ui/core/InputLabel";
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import FormControl from "@material-ui/core/FormControl";
+import MenuItem from "@material-ui/core/MenuItem";
+import { makeStyles } from "@material-ui/core/styles";
+import * as Actions from '../ducks/source';
+
+import 'react-datepicker/dist/react-datepicker-cssmodules.css';
+
+export function observingRunTitle(observingRun, instrumentList, telescopeList, groups) {
+  const { instrument_id } = observingRun;
+  const instrument = instrumentList?.filter((i) => i.id === instrument_id)[0];
+
+  const telescope_id = instrument?.telescope_id;
+  const telescope = telescopeList?.filter((t) => t.id === telescope_id)[0];
+
+  const group = groups?.filter((g) => g.id === observingRun.group_id)[0];
+
+  return `${observingRun?.calendar_date} ${instrument?.name}/${telescope?.nickname} (PI: ${observingRun?.pi} ${group?.name})`;
+}
+
+function makeMenuItem(observingRun, instrumentList, telescopeList, groups) {
```

Comment:
Refactor to React component

Reply:
Or simply move to the main return statement as it's so small

Reply:
Fixed